includes/admin/importers/class-wc-product-importer.php

```diff
@@ -175,6 +164,234 @@ public function handle_upload() {
 	}
 
 	/**
+	 * Read a CSV file.
+	 *
+	 * @param mixed $file
+	 * @param array $args See $default_args
+	 * @return array
+	 */
+	public function read_csv( $file, $args = array() ) {
+
+		$default_args = array(
+			'start_pos' => 0, // File pointer start.
+			'end_pos' => -1, // File pointer end.
+			'lines' => -1, // Max lines to read.
+			'mapping' => array(), // Column mapping. csv_heading => schema_heading.
+			'parse' => false, // Whether to sanitize and format data.
+		);
+		$args = wp_parse_args( $args, $default_args );
+
+		$data = array(
+			'raw_headers' => array(),
+			'data' => array()
+		);
+
+		if ( false !== ( $handle = fopen( $file, "r" ) ) ) {
+
+			$data['raw_headers'] = fgetcsv( $handle, 0, $this->delimiter );
+
+			if ( $args['start_pos'] != 0 ) {
+				fseek( $handle, (int) $args['start_pos'] );
+			}
+
+			while( false !== ( $row = fgetcsv( $handle, 0, $this->delimiter ) ) ) {
+				$data['data'][] = $row;
+
+	            if ( ( $args['end_pos'] > 0 && ftell( $handle ) >= $args['end_pos'] ) || 0 === --$args[
+	            	'lines'] ) {
+	            	break;
+				}
+			}
+		}
+
+		if ( ! empty( $args['mapping'] ) ) {
+			$data = $this->map_headers( $data, $args['mapping'] );
+		}
+
+		if ( $args['parse'] ) {
+			$data = $this->parse_data( $data );
+		}
+
+		return apply_filters( 'woocommerce_csv_product_import_data', $data, $file, $args );
+	}
+
+	/**
+	 * Map raw headers to known headers.
+	 *
+	 * @param array $data
+	 * @param array $mapping 'raw column name' => 'schema column name'
+	 * @return array
+	 */
+	public function map_headers( $data, $mapping ) {
+		$data['headers'] = array();
+		foreach ( $data['raw_headers'] as $heading ) {
+			$data['headers'] = isset( $mapping[ $heading ] ) ? $mapping[ $heading ] : $heading;
+		}
+		return $data;
+	}
+
+	/**
+	 * Map and format raw data to known fields.
+	 *
+	 * @param array $data
+	 * @return array
+	 */
+	public function parse_data( $data ) {
+
+		// Columns not mentioned here will get parsed with 'esc_attr'.
+		// column_name => callback
+		$data_formatting = array(
```

Comment:
No need to change it, but just a note for future reference: Maybe we need to make slugs from this keys or make translatable, since will be not good for non-English speakers.